catalogue/webapp/types/compressed-manifest.ts

```diff
@@ -0,0 +1,127 @@
+import { isNotUndefined } from '@weco/common/utils/type-guards';
+import {
+  longestCommonPrefix,
+  longestCommonSuffix,
+} from '@weco/common/utils/array';
+import { TransformedCanvas, TransformedManifest } from './manifest';
+
+type CompressedTransformedCanvases = {
+  idPrefix: string;
+  imageServiceIdPrefix: string;
+  labelPrefix: string;
+  textServiceIdPrefix: string;
+  thumbnailImageUrlPrefix: string;
+  thumbnailImageUrlSuffix: string;
```

Comment:
one area that needs testing: what if you send a single-canvases manifest? I suspect you'd get the entire URL as both prefix and suffix, which is bad. I feel like what this really wants is a function like:

```typescript
function getCommonParts(s: string[]): { prefix: string; suffix: string }
```

so that none of the common prefix ends up in the common suffix.

That could be used for the other prefixable-things, I think.